src/main/java/uk/gov/ons/ctp/response/action/scheduled/distribution/ActionDistributor.java

```diff
@@ -22,101 +21,102 @@
- */
+/** This is the service class that distributes actions to downstream services */
 @Component
 class ActionDistributor {
 
   private static final Logger log = LoggerFactory.getLogger(ActionDistributor.class);
 
-  @Autowired private AppConfig appConfig;
+  private static final String LOCK_PREFIX = "ActionDistributionLock-";
 
-  @Autowired private ActionRepository actionRepo;
+  private AppConfig appConfig;
+  private RedissonClient redissonClient;
 
-  @Autowired private ActionTypeRepository actionTypeRepo;
+  private ActionRepository actionRepo;
+  private ActionCaseRepository actionCaseRepo;
+  private ActionTypeRepository actionTypeRepo;
 
-  @Autowired
-  @Qualifier("business")
-  private ActionProcessingService businessActionProcessingService;
+  private CaseSvcClientService caseSvcClientService;
 
-  @Autowired
-  @Qualifier("social")
+  private ActionProcessingService businessActionProcessingService;
   private ActionProcessingService socialActionProcessingService;
 
-  @Autowired private ActionCaseRepository actionCaseRepo;
-
-  @Autowired private CaseSvcClientService caseSvcClientService;
+  public ActionDistributor(
+      AppConfig appConfig,
+      RedissonClient redissonClient,
+      ActionRepository actionRepo,
+      ActionCaseRepository actionCaseRepo,
+      ActionTypeRepository actionTypeRepo,
+      CaseSvcClientService caseSvcClientService,
+      @Qualifier("business") ActionProcessingService businessActionProcessingService,
+      @Qualifier("social") ActionProcessingService socialActionProcessingService) {
+    this.appConfig = appConfig;
+    this.redissonClient = redissonClient;
+    this.actionRepo = actionRepo;
+    this.actionCaseRepo = actionCaseRepo;
+    this.actionTypeRepo = actionTypeRepo;
+    this.caseSvcClientService = caseSvcClientService;
+    this.businessActionProcessingService = businessActionProcessingService;
+    this.socialActionProcessingService = socialActionProcessingService;
+  }
 
   /**
-   * wake up on schedule and check for submitted actions, enrich and distribute them to spring
-   * integration channels
-   *
-   * @return the info for the health endpoint regarding the distribution just performed
+   * Called on schedule to check for submitted actions then creates and distributes requests to
+   * action exporter or notify gateway
    */
   @Transactional
-  public DistributionInfo distribute() {
-    log.debug("ActionDistributor awoken...");
-    final DistributionInfo distInfo = new DistributionInfo();
-    final List<ActionType> actionTypes = actionTypeRepo.findAll();
-
-    for (final ActionType actionType : actionTypes) {
-      final List<InstructionCount> instructionCounts = processActionType(actionType);
-      distInfo.getInstructionCounts().addAll(instructionCounts);
-    }
-
-    log.debug("ActionDistributor going back to sleep");
-    return distInfo;
+  public void distribute() {
+    List<ActionType> actionTypes = actionTypeRepo.findAll();
```

Comment:
cheeky little improvement - a performance enhancement. If you create a wrapper service for calling this and mark the function with`@Cachable` then Spring knows that the ActionTypes don't change very often, so it'll give you a copy from an in-memory cache. I think there's an option which specifies how long the cache copy can live. Probably out of scope for this PR, but something worth bearing in mind if you're ever doing another Spring Boot app.